### PR TITLE
Fix Pandas compatibility

### PR DESCRIPTION
**Pull request by moprescu**
* Added array checks to ensure compatibility with Pandas
* Fixed failures when passing in column vectors for Y and T

### CODE REVIEW

File: econml/cate_estimator.py [hunk 1]
```diff
@@ -5,8 +5,10 @@
 
 import abc
 import numpy as np
+import pandas as pd
 from functools import wraps
 from copy import deepcopy
+from sklearn.utils import check_array
```
Comment (kbattocchi) [inline econml/cate_estimator.py:11]: The new imports here shouldn't be needed if there are no other changes to this file, right?
```suggestion
from functools import wraps
from copy import deepcopy
```

File: econml/dml.py [hunk 2]
```diff
@@ -696,6 +696,7 @@ def fit(self, Y, T, X=None, W=None, sample_weight=None, sample_var=None, inferen
         if sample_var is not None and inference is not None:
             warn("This estimator does not yet support sample variances and inference does not take "
                  "sample variances into account. This feature will be supported in a future release.")
+        Y, T, X, W, sample_weight, sample_var = check_input_arrays(Y, T, X, W, sample_weight, sample_var)
```
Comment (kbattocchi) [inline econml/dml.py:699]: Why only add the check for SparseLinearDML, but not the other classes in this file?

File: econml/drlearner.py [hunk 3]
```diff
@@ -928,6 +928,7 @@ def fit(self, Y, T, X=None, W=None, *, sample_weight=None, sample_var=None, infe
         if sample_weight is not None and inference is not None:
             warn("This estimator does not yet support sample variances and inference does not take "
                  "sample variances into account. This feature will be supported in a future release.")
+        Y, T, X, W, sample_weight, sample_var = check_input_arrays(Y, T, X, W, sample_weight, sample_var)
```
Comment (kbattocchi) [inline econml/drlearner.py:931]: Likewise, why only the SparseLinear version?  If it's because it's needed for the check on the next line, then maybe only run check_input_arrays on X and T and add a comment explaining.

File: econml/metalearners.py [hunk 4]
```diff
@@ -262,6 +262,8 @@ def fit(self, Y, T, X, *, inference=None):
         """
         # Check inputs
         Y, T, X, _ = check_inputs(Y, T, X, multi_output_T=False)
```
Comment (kbattocchi) [inline econml/metalearners.py:264]: Is check_inputs equivalent to check_input_arrays?  Would it be beneficial to switch things to the latter everywhere for consistency?

File: econml/metalearners.py [hunk 5]
```diff
@@ -262,6 +262,8 @@ def fit(self, Y, T, X, *, inference=None):
...
+        if Y.ndim == 2 and Y.shape[1] == 1:
+            Y = Y.flatten()
```
Comment (kbattocchi) [inline econml/metalearners.py:266]: Why is this necessary?  Please add a comment explaining, and a test that covers this scenario.

File: econml/utilities.py [hunk 6]
```diff
@@ -510,6 +511,12 @@ def check_inputs(Y, T, X, W=None, multi_output_T=True, multi_output_Y=True):
     return Y, T, X, W
 
 
+def check_input_arrays(*args):
+    # Check arrays for all sequence-like objects
```
Comment (kbattocchi) [inline econml/utilities.py:515]: Add proper docstring

File: econml/utilities.py [hunk 7]
```diff
@@ -4,6 +4,7 @@
 """Utility methods."""
 
 import numpy as np
+import pandas as pd
```
Comment (kbattocchi) [inline econml/utilities.py:7]: Doesn't appear necessary unless there are other changes.
```suggestion
```

**Review by kbattocchi — CHANGES_REQUESTED** [review]
Mostly looks fine, but definitely needs corresponding tests.

File: econml/dml.py (hunk 2 above)
Comment (moprescu) [reply econml/dml.py:699]: Because the sparse methods call other methods (e.g. `check_high_dimensional`) before calling the super's fit. All other DML and DR estimators call the parent's fit (`_ortho_learner` in this case).

File: econml/drlearner.py (hunk 3 above)
Comment (moprescu) [reply econml/drlearner.py:931]: Same as above.

File: econml/utilities.py (hunk 6 above)
Comment (moprescu) [reply econml/utilities.py:534]: Done!

**Review by kbattocchi — APPROVED** [review]

File: econml/metalearners.py (hunk 5 above)
Comment (moprescu) [reply econml/metalearners.py:266]: The `sklearn` models flatten the output and the difference between an `(n, 1)` and an `(n, )`  is an `(n, n)` array. Since there are several models where we compute differences this way and it's difficult to keep track of the shapes, it's more straightforward to just flatten the input `Y`, I'll make sure to add a comment that reflects this.
